tests/ut/azure_iot_hub_client_properties_ut.c

```diff
@@ -513,6 +581,140 @@ static void testAzureIoTHubClientProperties_GetNextComponentProperty_Success( vo
 }
 
+static void testAzureIoTHubClientProperties_GetNextComponentPropertyGetDocument_Success( void ** ppvState )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+    AzureIoTJSONReader_t xJSONReader;
+    AzureIoTJSONTokenType_t xTokenType;
+    AzureIoTHubMessageType_t xResponseType = eAzureIoTHubPropertiesGetMessage;
+    AzureIoTHubClientPropertyType_t xPropertyType = eAzureIoTHubClientPropertyWritable;
+    uint32_t ulVersion;
+    const uint8_t * ucComponentName;
+    uint32_t ulComponentNameLength;
+    int32_t lValue = 0;
+
+    AzureIoTHubClientOptions_t xOptions;
+
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTSuccess );
+
+    AzureIoTHubClient_OptionsInit( &xOptions );
+
+    AzureIoTHubClient_Init( &xTestIoTHubClient,
+                            ucHostname,
+                            strlen( ucHostname ),
+                            ucDeviceId,
+                            strlen( ucDeviceId ),
+                            &xOptions,
+                            ucBuffer, sizeof( ucBuffer ),
+                            prvGetUnixTime,
+                            &xTransportInterface );
+
+    assert_int_equal( AzureIoTJSONReader_Init(
+                          &xJSONReader,
+                          ucTestJSONGetPayload,
+                          strlen( ucTestJSONGetPayload ) ),
+                      eAzureIoTSuccess );
+
+    /* First component | first property */
+    assert_int_equal( AzureIoTHubClientProperties_GetNextComponentProperty( &xTestIoTHubClient,
+                                                                            &xJSONReader,
+                                                                            xResponseType,
+                                                                            xPropertyType,
+                                                                            &ucComponentName,
+                                                                            &ulComponentNameLength ), eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoTJSONReader_TokenType( &xJSONReader, &xTokenType ), eAzureIoTSuccess );
+    assert_int_equal( xTokenType, eAzureIoTJSONTokenPROPERTY_NAME );
+    assert_true( AzureIoTJSONReader_TokenIsTextEqual( &xJSONReader, "targetTemperature", strlen( "targetTemperature" ) ) );
+    assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
+    /*Advance to property value */
+    assert_int_equal( AzureIoTJSONReader_TokenType( &xJSONReader, &xTokenType ), eAzureIoTSuccess );
+    assert_int_equal( xTokenType, eAzureIoTJSONTokenNUMBER );
+    assert_int_equal( AzureIoTJSONReader_GetTokenInt32( &xJSONReader, &lValue ), eAzureIoTSuccess );
+    assert_int_equal( lValue, 40 );
+    assert_int_equal( AzureIoTJSONReader_NextToken( &xJSONReader ), eAzureIoTSuccess );
+
+    /* End of properties */
+    assert_int_equal( AzureIoTHubClientProperties_GetNextComponentProperty( &xTestIoTHubClient,
```

Comment:
how about showing things working in while loop, so as to use this API like iterator? I know here we are unwrapping loop, but having a UT showing that will be good to have.

Reply:
I can add a backlog item to do this since I think it's a good idea. I don't want to spend the cycles right now doing it though since I really just need these updates for the sample